aten/src/ATen/native/Loss.cpp

```diff
@@ -70,6 +74,76 @@ TORCH_IMPL_FUNC(smooth_l1_loss_out)
   }
 }
 
+Tensor& margin_ranking_loss_out
+(const Tensor& input1,
+ const Tensor& input2,
+ const Tensor& target,
+ double margin,
+ int64_t reduction,
+ Tensor& result
+) {
+  TORCH_CHECK(
+    !input1.is_complex() &&
+    !input2.is_complex() &&
+    !target.is_complex(),
+    "margin_ranking_loss expects inputs to be non-complex");
+
+  // Disallow broadcasting:
+  // https://github.com/pytorch/pytorch/pull/64975
+  TORCH_CHECK(
+    input1.sizes() == input2.sizes() &&
+    input1.sizes() == target.sizes(),
+    "margin_ranking_loss: all input tensors should have same size but got sizes"
+    ": input1: ", input1.sizes(),
+    ", input2: ", input2.sizes(),
+    ", target: ", target.sizes());
+
+  const auto common_type1 = promoteTypes(input1.scalar_type(), target.scalar_type());
```

Comment:
why are you doing manual type promotion instead of letting TI take care of it?

Reply:
I didn't know that that could be done. Can you simply set `result` to be of the same type of, say `input1`,  and TI will take care of cast it to the correct type?

Reply:
Undefined result will be set to the common type of inputs, dtype of defined result will be respected, but computation will still be done in the common dtype of inputs and copied to result.

Reply:
I am not 100% sure, as I also remember to have problems passing an undefined tensor to TI. Perhaps @ngimel can help here?

Reply:
okay i'll take a look, ty